frontend/src/core/comments/components/AddComment.test.js

```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+import { shallow } from 'enzyme';
+import sinon from 'sinon';
+
+import AddComment from './AddComment';
+
+describe('<AddComment>', () => {
+    it('calls submitComment function', () => {
+        const submitCommentFn = sinon.spy();
+        const wrapper = shallow(<AddComment
+            submitComment={ submitCommentFn }
+        />);
+
+        wrapper.prop('onSubmit') === submitCommentFn;
```

Comment:
So this component has a Form with an `onSubmit` that calls the `submitComment` function. My attempts to test the submit event have been unsuccessful.  The main error I get is: `Method “simulate” is meant to be run on 1 node. 0 found instead.` 

I've tried to find it by 'input' and 'form' as well as by 'id' all with the same result.

Reply:
That means there was a problem with either the component or the selector you wrote, I guess. Can you find or write back that code and show it to us?

Reply:
You can trigger the error by replacing the line above with:
```js
wrapper.find('form').simulate('submit', { submitCommentFn });
```

I guess you hit the `return null;` in the component, because you don't set the `user` prop, hence the `<form>` is never rendered.